Modules/_sqlite/cursor.c

```diff
@@ -426,6 +426,7 @@ PyObject* _pysqlite_query_execute(pysqlite_Cursor* self, int multiple, PyObject*
         }
     } else {
         /* execute() */
+        second_argument = _PyTuple_Empty;
         if (!PyArg_ParseTuple(args, "O|O", &operation, &second_argument)) {
             goto error;
         }
```

Comment:
_pysqlite_query_execute() is an implementation of two different but similar methods: execute() (multiple=0) and executemany() (multiple=1). The second argument of executemany() is an iterable, some operation is executed for every item from that iterable. The second argument of execute() is optional, some operation is executed for it if it is specified or for empty tuple if it is missed.

I think the code of _pysqlite_query_execute() can be simplified even more, but this is a different issue, not directly related to empty tuples.